app/workers/preparsing.rb

```diff
@@ -10,11 +12,40 @@ class Preparsing
   def perform(genotype_id)
     genotype = Genotype.find(genotype_id)
 
-    logger.info "Starting preparse"
-    biggest = ''
-    biggest_size = 0
-    begin
-      Zip::File.open(genotype.genotype.path) do |zipfile|
+    logger.info "Starting preparse on #{genotype.genotype.path}"
+    # First, we need to find out which archive or flat text our uploaded file is!
+    # We use the bash tool file for that
+    #
+    # There are two possible outcomes - file is a collection of files (tar, tar.gz, zip)
+    # or file is a single file (ASCII, gz)
+    filetype = `file #{genotype.genotype.path}`
```

Comment:
I saw some `.docx` files as well.. not sure if after extraction or before..

Reply:
Haha, yeah, some folks aren't great at uploading the right file types. I'd say we should reject anything that's not ASCII post-unzip :)

Reply:
That's true. I also saw a few post-genotype analyses from Promethease.